### PR TITLE
fix: crash when converting infinite values to JSON string

### PR DESCRIPTION
**Pull request by zhu-xiaowei**
## Description
1. fix the issue with converting non-finite values to JSON string

## General Checklist

- [x] Added new tests to cover change, if needed
- [x] Build succeeds using Swift Package Manager
- [x] All unit tests pass
- [ ] Security oriented best practices and standards are followed (e.g. using input sanitization, principle of least privilege, etc)
- [ ] Documentation update for the change if required
- [x] PR title conforms to conventional commit style
- [ ] If breaking change, documentation/changelog update with migration instructions

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: Sources/Clickstream/Dependency/Clickstream/Event/Event.swift [hunk 1]
```diff
@@ -93,6 +93,7 @@ enum Event {
         static let ATTRIBUTE_NAME_INVALID = 2_002
         static let ATTRIBUTE_VALUE_LENGTH_EXCEED = 2_003
         static let ATTRIBUTE_SIZE_EXCEED = 2_004
+        static let ATTRIBUTE_VALUE_NOT_FINITE = 2_005
```
Comment (zxkane) [inline Sources/Clickstream/Dependency/Clickstream/Event/Event.swift:96]: change to infinite

File: Sources/Clickstream/Dependency/Clickstream/Event/EventChecker.swift [hunk 2]
```diff
@@ -82,13 +82,35 @@ class EventChecker {
                 error.errorCode = Event.ErrorCode.ATTRIBUTE_VALUE_LENGTH_EXCEED
                 error.errorMessage = getErrorMessage(errorString)
             }
+        } else if !checkFinite(value) {
+            errorMsg = """
+            the value for attribute : \(key), is not finite\
+             and the attribute will not be recorded
+            """
+            let errorString = "the value for attribute name: \(key) is not finite"
```
Comment (zxkane) [inline Sources/Clickstream/Dependency/Clickstream/Event/EventChecker.swift:90]: use infinite

**Review by zxkane — CHANGES_REQUESTED** [review]

File: Sources/Clickstream/Dependency/Clickstream/Event/Event.swift (hunk 1 above)
Comment (zhu-xiaowei) [reply Sources/Clickstream/Dependency/Clickstream/Event/Event.swift]: fixed

File: Sources/Clickstream/Dependency/Clickstream/Event/EventChecker.swift (hunk 2 above)
Comment (zhu-xiaowei) [reply Sources/Clickstream/Dependency/Clickstream/Event/EventChecker.swift]: done.

**Review by zxkane — APPROVED** [review]

**Pull request merged**